app/controllers/users.js

```diff
@@ -19,6 +26,23 @@ const createUser = async (req, res, next) => {
   }
 };
 
-module.exports = {
-  createUser
+exports.signIn = async (req, res, next) => {
+  try {
+    const { mail, password } = req.body;
+    const user = await UserServices.getUserByEmail(mail);
+    if (!user) return next(unauthorized(BAD_CREDENTIALS));
+    const validateData = await comparePassword(password, user.password);
+    if (validateData) {
+      const { id, name, lastName } = user;
+      const token = generateToken({ id, name, lastName, mail });
+      return res.status(200).send({
+        message: SIGN_IN_SUCCESSFUL,
+        data: { token }
+      });
+    }
+    return next(unauthorized(BAD_CREDENTIALS));
+  } catch (error) {
+    logger.error(error);
+    return next(serviceError(SIGN_IN_MESSAGE_ERROR));
```

Comment:
Excuse me, this also escaped me, but to the `next` function only pass the error, if you leave it like that, it would be replacing the error that was thrown in the other operations

Reply:
¿Is this comment just referred to line 46?